liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/PluginConfigSupport.java

```diff
@@ -203,10 +203,37 @@ protected boolean isAppConfigInSourceServerXml() {
         
         if (serverXML != null && serverXML.exists()) {
             try {
-                bConfigured = ServerXmlDocument.isFoundWebApplication(serverXML.getCanonicalPath(), configDirectory);
+                ServerConfigDocument scd = ServerConfigDocument.getInstance(serverXML, configDirectory);
+                
+                if (scd != null && scd.getLocations().contains(fileName)) {
+                    log.debug("Application configuration is found in server.xml : " + fileName);
+                    bConfigured = true;
+                }
             } 
             catch (Exception e) {
-                log.debug("Exception is thrown by ServerXmlDocument.isFoundWebApplication : " + e);
+                log.debug("Exception is thrown by ServerConfigDocument.isAppConfiguredInSourceServerXml : " + e);
```

Comment:
If things like malformed XML can cause this exception, would a warning be better?